core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala

```diff
@@ -704,6 +707,43 @@ class MasterSuite extends SparkFunSuite
   private def getState(master: Master): RecoveryState.Value = {
     master.invokePrivate(_state())
   }
+
+  test("Total cores is not divisible by cores per executor") {
```

Comment:
IIUC I don't think this UT really reflects the code you changed in `Master`, even with the original code this two UTs should also be passed. I think you should test if the logics you changed is executed or not.